sample-prep/Biomek_pooling_VI.py

```diff
@@ -29,107 +43,287 @@ def main(process_id, file_id):
 
     lims.get_batch(inputs + outputs)
 
-    #load variables needed for script
+    # LOAD DATA required for the generated files
+    # calculating sample volumes from molMbp process udf and input udfs Genome Size and Molarity
+    
+    MAX_TOTAL_VOLUME = 1500
+    
+    sample_volumes = []
+    source_location = []
+    sample_name = []
+    molarity = []
+    project_account = []
+    genome_size = []
+    species = []
+    input_container_type =[]
+    input_container_name =[]
 
-    sample_volumes = [round(input.udf.get('Sample volume to pool (µl)'),2) for input in inputs]
-    source_location = [input.location for input in inputs]
-    Pool_names = [output.name for output in outputs]
-    sample_name = [input.name for input in inputs]
-    molarity = [input.udf.get('Molarity TS (nM)') for input in inputs]
-    project_account = [input.udf.get('Project Account') for input in inputs]
-    genome_size = [input.udf.get('Genome Size (Mbp)') for input in inputs]
-    species = [input.udf.get('Species') for input in inputs]
-    total_sample_number = len(sample_name)
+    molMbp = process.udf.get('fmol/Mbp')
+    min_vol_required = process.udf.get('Min Volume Required')
+    max_vol_available = process.udf.get('Max Volume Available')
+    enforce_max_vol = process.udf.get('Enforce Max Volume Available')
+    enforce_min_volume = process.udf.get('Enforce Min Volume Required')
+    
+    for input in sorted(inputs, key=sort_key):
+        source_location.append(input.location)
+        sample_name.append(input.name)
+        molarity.append(input.udf.get('Molarity TS (nM)'))
+        project_account.append(input.udf.get('Project Account'))
+        genome_size.append(input.udf.get('Genome Size (Mbp)'))
+        species.append(input.udf.get('Species'))
+        input_container_type.append(input.container.type.name)
+        input_container_name.append(input.container.name)
+        sample_volumes.append(round(molMbp*input.udf.get('Genome Size (Mbp)')/input.udf.get('Molarity TS (nM)'),2))
 
-    #exit script if entered step with more than one pool
+    # loop for 'Enforce Min Volume Required' button on the step. 
+    # If true, find the minimum value, find its index.
+    # Recalculate molMbp to achieve min_vol_required. New value put to the process udf for info to the user. volume_threshold_plotcolor set to yellow
+    # finally a for-loop to update sample_volumes using the new fmol/Mbp
```

Comment:
You might want to include some more context here, mostly to be kind to yourself next year :) Not just what is happening, but also why.

Reply:
I think this might be a bit more clear in the new version (https://github.com/NorwegianVeterinaryInstitute/nvi_lims_epps)